### PR TITLE
Bugfix: Encounter follow ups don't show up on same day

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [ch4414](https://app.clubhouse.io/simpledotorg/story/4414/et-zeros-in-progress-tab)

## Because

Daily follow up numbers on the progress tab number don't show up on the same day.

## This addresses

When a facility has both HTN and DM enabled, we fetch daily numbers by comparing `encounters.encountered_on`. However this is a date field, not a datetime field, and the date range comparison excludes today's results. This happens because groupdate tries to convert the time zone of the date. The [recommended way](https://github.com/ankane/groupdate#date-columns)  to avoid the time zone conversion is to pass in a  `time_zone: false` to groupdate.

### CODE REVIEW

File: app/models/patient.rb
Comment (harimohanraj89) [inline app/models/patient.rb:82]: The `time_zone: false` is the crux of the bugfix. The rest is cleanup and refactoring.

File: app/queries/follow_ups_query.rb
Comment (harimohanraj89) [inline app/queries/follow_ups_query.rb:12]: The key change here is accepting a blob of options to pass transparently to groupdate.

File: spec/queries/follow_ups_query_spec.rb
Comment (rsanheim) [inline spec/queries/follow_ups_query_spec.rb:133]: Can you also freeze time here? Otherwise this will fail around day boundaries for certain time zones.

File: app/queries/follow_ups_query.rb
Comment (rsanheim) [inline app/queries/follow_ups_query.rb:5]: 👍

**Review by rsanheim — APPROVED** [review]
Approving to unblock, but one thing about making the test consistent.

**Comment by harimohanraj89** [comment]
@claudiovallejo you are tagged since you're a codeowner and this touches an indicator's query. I'm dismissing your review since this is a bugfix, and no definitions are being changed.

Good to know this thing works!

**Pull request merged**
